ding/worker/vllm_test_wqj.py

```diff
@@ -0,0 +1,222 @@
+        self.temperature = temperature
+
+    async def generate(
+        self,
+        prompt,
+        num_samples: int,
+    ) -> List[Tuple[str, float]]:
+        """
+        Overview:
+            Generate tactics for the current state.
+        Arguments:
+            - prompt : The prompt to generate tactics.
+            - num_samples (int): The number of tactics to generate.
+        Returns:
+            - List[Tuple[str, float]]: The generated tactics and their log-probabilities.
+
+        .. note::
+            This method is asynchronous and returns a coroutine.
+        """
+        response = await self.vllm_actor.generate(prompt, num_samples, self.max_tokens, self.temperature)
+        # Use raw logprobs as confidence scores
+        confidence_scores = [x.cumulative_logprob for x in response.outputs]
+        return [
+            (x.text.strip(), conf)
+            for x, conf in zip(response.outputs, confidence_scores)
+        ]
+        
+
+model=HuggingFaceModelGenerator('/mnt/afs/share/Qwen2-VL-7B',temperature=0.5) #设置一个temperature就好了,可以做到生成多个候选答案
+
+def get_prompts_qwen(questions: list,modality: str):
+    if modality == "image":
+        placeholder = "<|image_pad|>"
+    elif modality == "video":
+        placeholder = "<|video_pad|>"
+
+    prompts = [("<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n"
+              f"<|im_start|>user\n<|vision_start|>{placeholder}<|vision_end|>"
+              f"{question}<|im_end|>\n"
+              "<|im_start|>assistant\n") for question in questions]
+    stop_token_ids = None
+    return prompts,stop_token_ids
+
+def get_multi_modal_input(modality,filenames,questions):
+    """
+    return {
+        "data": image or video,
+        "question": question,
+    }
+    """
+    if modality == "image":
+        # Input image and question
+        ret={}
+        ret["data"]=[]
+        ret["question"]=[]
+        for filename,question in zip(filenames,questions):
+            image = ImageAsset(filename) \
+                .pil_image.convert("RGB")
+            #img_question = "What is the content of this image?"
+            img_question=question
+            ret["data"].append(image)
+            ret["question"].append(img_question)
+        return ret
+
+
+questions=["What is the content of this image?","Please describe the image.","How many people are there in the image? What are they doing?"]
+img_names=['/mnt/afs/niuyazhe/data/meme/data/Eimages/Eimages/Eimages/image_ (2)','/mnt/afs/niuyazhe/data/meme/data/Cimages/Cimages/Cimages/Image_(3127)','/mnt/afs/wangqijian/data/test/test']
+#questions=["What is the content of this image?"]
+#img_names=['/mnt/afs/niuyazhe/data/meme/data/Eimages/Eimages/Eimages/image_ (2)']
+num_prompts=len(questions)
+image_repeat_prob=None
+modality = 'image'
+
+mm_input = get_multi_modal_input(modality,img_names,questions)
+data = mm_input["data"]
+question = mm_input["question"]
+batch_inference_mine=True
```

Comment:
what is the usage of this argument, check all the similar cases, if redundant, please remove them, if these arguments will be used in the future, please add some comments to explain their meanings and goals.